### PR TITLE
fix: Update model checkpoint kwargs

### PR DESCRIPTION
**Pull request by ktagowski**

### CODE REVIEW

File: embeddings/pipeline/lightning_pipeline.py [hunk 1]
```diff
@@ -27,6 +27,11 @@ class LightningPipeline(
         "mode": "min",
         "patience": 3,
     }
+    DEFAULT_MODEL_CHECKPOINT_KWARGS: Dict[str, Any] = {
+        "filename": "last",
+        "monitor": None,
+        "save_last": False,
+    }
```
Comment (djaniak) [inline embeddings/pipeline/lightning_pipeline.py:34]: Do we only want to save the last epoch checkpoint? what if the model will start to overfit heavily?

File: embeddings/pipeline/lightning_sequence_labeling.py [hunk 2]
```diff
@@ -38,10 +38,16 @@ def __init__(
...
+        model_checkpoint_kwargs: Optional[Dict[str, Any]] = None,
     ):
         task_train_kwargs = config.task_train_kwargs
         task_train_kwargs.update({"devices": devices, "accelerator": accelerator})
         tokenizer_name_or_path = tokenizer_name_or_path or embedding_name_or_path
+        model_checkpoint_kwargs = (
+            model_checkpoint_kwargs
+            if model_checkpoint_kwargs
+            else LightningPipeline.DEFAULT_MODEL_CHECKPOINT_KWARGS
+        )
```
Comment (asawczyn) [inline embeddings/pipeline/lightning_sequence_labeling.py:50]: if user passes kwargs (e.g. `{"save_last": True}`), all of the default kwargs are overwritten. Is it intentional?

Comment (asawczyn) [reply embeddings/pipeline/lightning_sequence_labeling.py:50]: Moreover if the passed dict is empty i.e.  `{}`, it is overwritten by default kwargs.

**Review by asawczyn — CHANGES_REQUESTED** [review]

File: embeddings/pipeline/lightning_pipeline.py (hunk 1 above)
Comment (ktagowski) [reply embeddings/pipeline/lightning_pipeline.py:34]: We have early stopping as default in our pipelines

File: embeddings/pipeline/lightning_sequence_labeling.py (hunk 2 above)
Comment (ktagowski) [reply embeddings/pipeline/lightning_sequence_labeling.py]: Fixed

Comment (asawczyn) [reply embeddings/pipeline/lightning_sequence_labeling.py]: Do you use `DEFAULT_MODEL_CHECKPOINT_KWARGS` now?

Comment (ktagowski) [reply embeddings/pipeline/lightning_sequence_labeling.py]: Currently it's the same behaviour as with other `DEFAULT_KWARGS` User can use it, but automatically is not set.

Comment (ktagowski) [reply embeddings/pipeline/lightning_sequence_labeling.py]: removed

**Review by asawczyn — APPROVED** [review]

**Pull request merged**